future/src/ct/ct_export2pdf.cc

```diff
@@ -325,7 +327,7 @@ void CtPrint::_on_draw_page_text(const Glib::RefPtr<Gtk::PrintContext>& context,
 {
     // layout num, line num
     std::pair<int, int> start_line_num = page_nr == 0 ? std::make_pair(0, 0) : print_data->page_breaks[page_nr - 1];
-    std::pair<int, int> end_line_num = page_nr < (int)print_data->page_breaks.size() ? print_data->page_breaks[page_nr] : std::make_pair((int)print_data->layout.size()-1, print_data->layout_num_lines.back());
+    std::pair<int, int> end_line_num = page_nr < static_cast<int>(print_data->page_breaks.size()) ? print_data->page_breaks[page_nr] : std::make_pair(static_cast<int>(print_data->layout.size()-1), print_data->layout_num_lines.back());
```

Comment:
that is why I hate unsigned int. Are you sure it will be ok when layout.size() == 0?
std::make_pair(static_cast<int>(print_data->layout.size()-1)

Reply:
Sorry my comment did not appear, yes this will cast properly (so `0 - 1` == `0` instead of wrapping)